src/ray/gcs/gcs_server/gcs_resource_manager.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+#include "ray/gcs/gcs_server/gcs_node_manager.h"
+
+namespace ray {
+namespace gcs {
+
+/// Gcs resource manager interface.
+/// It is used for actor and placement group scheduling.
+/// Non-thread safe.
+class GcsResourceManagerInterface {
+ public:
+  virtual ~GcsResourceManagerInterface() {}
+
+  /// Get the resources of all nodes in the cluster.
+  ///
+  /// \return The resources of all nodes in the cluster.
+  virtual const absl::flat_hash_map<NodeID, ResourceSet> &GetClusterResources() const = 0;
```

Comment:
Why not use `const absl::flat_hash_map<NodeID, SchedulingResources> &`

Reply:
The resources of actor / task are not managed in GCS. We can only get the available resources from heartbeat.